src/Agent/NewRelic/Agent/Core/Agent.cs

```diff
@@ -292,60 +293,25 @@ public void HandleWrapperException(Exception exception)
 
         public Stream TryGetStreamInjector(Stream stream, Encoding encoding, string contentType, string requestPath)
         {
-            if (stream == null)
-            {
-                return null;
-            }
-
-            if (encoding == null)
-            {
-                return null;
-            }
-
-            if (contentType == null)
-            {
-                return null;
-            }
-
-            if (requestPath == null)
-            {
-                return null;
-            }
-
-            try
-            {
-                var transaction = _transactionService.GetCurrentInternalTransaction();
-                if (transaction == null)
-                {
-                    return null;
-                }
-
-                var shouldInject = _browserMonitoringPrereqChecker.ShouldAutomaticallyInject(transaction, requestPath, contentType);
-                if (!shouldInject)
-                {
-                    return null;
-                }
+            var script = TryGetRUMScriptInternal(contentType, requestPath);
+            return script == null ? null : new BrowserMonitoringStreamInjector(() => script, stream, encoding);
```

Comment:
Do we need to use a lambda to pass the script to the stream injector? It seems like we could reduce allocations if we just pass the script directly.

Reply:
The lambda function is to accommodate unit testing.

Reply:
Is there another way to write the tests so that we do not need the extra allocations and overhead in the agent code?

Reply:
I'm not sure; I was just replicating the way `BrowserMonitoringWriter` and its unit tests (in `RumLoaderInsertionLocationTests`) was written. The only real difference between the classes is that mine is `static`.

Reply:
It just seems like complexity that we don't need. I could see us passing a lambda or local function if we needed to lazy load, or defer execution of some logic. However, in this case, we already executed the code to get the script, so it doesn't seem necessary to wrap it in a lambda or local function.

Reply:
Turns out it was pretty easy to refactor and just pass the byte array directly.